module-system/module-implementations/sov-attester-incentives/src/call.rs

```diff
@@ -329,62 +345,46 @@ impl<
+                .rollup_finality_period
+                .get(working_set)
+                .expect("Should be defined at genesis");
+
+            if TransitionHeight(
+                unbonding_info
+                    .unbonding_initiated_height
+                    .inner()
+                    .saturating_add(finality_period.inner()),
+            ) > curr_height
             {
-                // These two constants should always be set beforehand, hence we can panic if they're not set
-                let curr_height = self
-                    .light_client_finalized_height
-                    .get(working_set)
-                    .expect("Should be defined at genesis");
-                let finality_period = self
-                    .rollup_finality_period
-                    .get(working_set)
-                    .expect("Should be defined at genesis");
-
-                if begin_unbond_height + finality_period > curr_height {
-                    return Err(self.slash_burn_reward(
-                        context.sender(),
-                        Role::Attester,
-                        SlashingReason::UnbondingNotFinalized,
-                        working_set,
-                    ));
-                }
-
-                // Get the user's old balance.
-                if let Some(old_balance) = self.bonded_attesters.get(context.sender(), working_set)
-                {
-                    // Transfer the bond amount from the sender to the module's address.
-                    // On failure, no state is changed
-                    self.reward_sender(context, old_balance, working_set)?;
-
-                    // Update our internal tracking of the total bonded amount for the sender.
-                    self.bonded_attesters.remove(context.sender(), working_set);
-                    self.unbonding_attesters
-                        .remove(context.sender(), working_set);
-
-                    // Emit the unbonding event
-                    working_set.add_event(
-                        "unbonded_challenger",
-                        &format!("amount_withdrawn: {old_balance:?}"),
-                    );
-                }
-            } else {
-                return Err(self.slash_burn_reward(
-                    context.sender(),
-                    Role::Attester,
-                    SlashingReason::AttesterIsNotUnbonding,
-                    working_set,
-                ));
+                return Err(AttesterIncentiveErrors::UnbondingNotFinalized);
             }
+
+            // Get the user's old balance.
+            // Transfer the bond amount from the sender to the module's address.
+            // On failure, no state is changed
+            self.reward_sender(context, unbonding_info.amount, working_set)?;
+
+            // Update our internal tracking of the total bonded amount for the sender.
+            self.bonded_attesters.remove(context.sender(), working_set);
```

Comment:
I think this remove is redundant. We should have already removed the sender when they initiated the unbonding process

Reply:
Done